# Visualización de datos - Practica 2: Creación de visualización

In [1]:
import pandas as pd
import numpy as np
import seaborn as sb

In [2]:
# Se leerán los datos a utilizar para la visualización

data = pd.read_csv("winemag-data-130k-v2.csv", index_col=0)
data.head(10)

country                                        description  \
0     Italy  Aromas include tropical fruit, broom, brimston...   
1  Portugal  This is ripe and fruity, a wine that is smooth...   
2        US  Tart and snappy, the flavors of lime flesh and...   
3        US  Pineapple rind, lemon pith and orange blossom ...   
4        US  Much like the regular bottling from 2012, this...   
5     Spain  Blackberry and raspberry aromas show a typical...   
6     Italy  Here's a bright, informal red that opens with ...   
7    France  This dry and restrained wine offers spice in p...   
8   Germany  Savory dried thyme notes accent sunnier flavor...   
9    France  This has great depth of flavor with its fresh ...   

                          designation  points  price           province  \
0                        Vulkà Bianco      87    NaN  Sicily & Sardinia   
1                            Avidagos      87   15.0              Douro   
2                                 NaN      87   14.0             Oregon   
3                Reserve Late Harvest      87   13.0           Michigan   
4  Vintner's Reserve Wild Child Block      87   65.0             Oregon   
5                        Ars In Vitro      87   15.0     Northern Spain   
6                             Belsito      87   16.0  Sicily & Sardinia   
7                                 NaN      87   24.0             Alsace   
8                               Shine      87   12.0        Rheinhessen   
9                         Les Natures      87   27.0             Alsace   

              region_1           region_2         taster_name  \
0                 Etna                NaN       Kerin O’Keefe   
1                  NaN                NaN          Roger Voss   
2    Willamette Valley  Willamette Valley        Paul Gregutt   
3  Lake Michigan Shore                NaN  Alexander Peartree   
4    Willamette Valley  Willamette Valley        Paul Gregutt   
5              Navarra                NaN   Michael Schachner   
6             Vittoria                NaN       Kerin O’Keefe   
7               Alsace                NaN          Roger Voss   
8                  NaN                NaN  Anna Lee C. Iijima   
9               Alsace                NaN          Roger Voss   

  taster_twitter_handle                                              title  \
0          @kerinokeefe                  Nicosia 2013 Vulkà Bianco  (Etna)   
1            @vossroger      Quinta dos Avidagos 2011 Avidagos Red (Douro)   
2           @paulgwine       Rainstorm 2013 Pinot Gris (Willamette Valley)   
3                   NaN  St. Julian 2013 Reserve Late Harvest Riesling ...   
4           @paulgwine   Sweet Cheeks 2012 Vintner's Reserve Wild Child...   
5           @wineschach  Tandem 2011 Ars In Vitro Tempranillo-Merlot (N...   
6          @kerinokeefe   Terre di Giurfo 2013 Belsito Frappato (Vittoria)   
7            @vossroger              Trimbach 2012 Gewurztraminer (Alsace)   
8                   NaN  Heinz Eifel 2013 Shine Gewürztraminer (Rheinhe...   
9            @vossroger  Jean-Baptiste Adam 2012 Les Natures Pinot Gris...   

              variety               winery  
0         White Blend              Nicosia  
1      Portuguese Red  Quinta dos Avidagos  
2          Pinot Gris            Rainstorm  
3            Riesling           St. Julian  
4          Pinot Noir         Sweet Cheeks  
5  Tempranillo-Merlot               Tandem  
6            Frappato      Terre di Giurfo  
7      Gewürztraminer             Trimbach  
8      Gewürztraminer          Heinz Eifel  
9          Pinot Gris   Jean-Baptiste Adam

In [3]:
# Se obtienen las diferentes features que compenen el set de datos
data.columns

Index(['country', 'description', 'designation', 'points', 'price', 'province',
       'region_1', 'region_2', 'taster_name', 'taster_twitter_handle', 'title',
       'variety', 'winery'],
      dtype='object')

In [4]:
# Se obtiene la longitud total del set de datos y el número de features
print(data.shape)

(129971, 13)


In [5]:
# Obtenemos los números de los valores nulos para cada columna en el dataset
missing_values_count = data.isnull().sum() 
missing_values_count

country                     63
description                  0
designation              37465
points                       0
price                     8996
province                    63
region_1                 21247
region_2                 79460
taster_name              26244
taster_twitter_handle    31213
title                        0
variety                      1
winery                       0
dtype: int64

In [6]:
data_drop = data.drop(columns=['region_1', 'region_2', 'designation', 'province'])
print(data_drop.shape)
print("\n\n\n")
data_drop.head(20)

(129971, 9)






country                                        description  points  \
0       Italy  Aromas include tropical fruit, broom, brimston...      87   
1    Portugal  This is ripe and fruity, a wine that is smooth...      87   
2          US  Tart and snappy, the flavors of lime flesh and...      87   
3          US  Pineapple rind, lemon pith and orange blossom ...      87   
4          US  Much like the regular bottling from 2012, this...      87   
5       Spain  Blackberry and raspberry aromas show a typical...      87   
6       Italy  Here's a bright, informal red that opens with ...      87   
7      France  This dry and restrained wine offers spice in p...      87   
8     Germany  Savory dried thyme notes accent sunnier flavor...      87   
9      France  This has great depth of flavor with its fresh ...      87   
10         US  Soft, supple plum envelopes an oaky structure ...      87   
11     France  This is a dry wine, very spicy, with a tight, ...      87   
12         US  Slightly reduced, this wine offers a chalky, t...      87   
13      Italy  This is dominated by oak and oak-driven aromas...      87   
14         US  Building on 150 years and six generations of w...      87   
15    Germany  Zesty orange peels and apple notes abound in t...      87   
16  Argentina  Baked plum, molasses, balsamic vinegar and che...      87   
17  Argentina  Raw black-cherry aromas are direct and simple ...      87   
18      Spain  Desiccated blackberry, leather, charred wood a...      87   
19         US  Red fruit aromas pervade on the nose, with cig...      87   

    price         taster_name taster_twitter_handle  \
0     NaN       Kerin O’Keefe          @kerinokeefe   
1    15.0          Roger Voss            @vossroger   
2    14.0        Paul Gregutt           @paulgwine    
3    13.0  Alexander Peartree                   NaN   
4    65.0        Paul Gregutt           @paulgwine    
5    15.0   Michael Schachner           @wineschach   
6    16.0       Kerin O’Keefe          @kerinokeefe   
7    24.0          Roger Voss            @vossroger   
8    12.0  Anna Lee C. Iijima                   NaN   
9    27.0          Roger Voss            @vossroger   
10   19.0      Virginie Boone               @vboone   
11   30.0          Roger Voss            @vossroger   
12   34.0      Virginie Boone               @vboone   
13    NaN       Kerin O’Keefe          @kerinokeefe   
14   12.0       Matt Kettmann         @mattkettmann   
15   24.0  Anna Lee C. Iijima                   NaN   
16   30.0   Michael Schachner           @wineschach   
17   13.0   Michael Schachner           @wineschach   
18   28.0   Michael Schachner           @wineschach   
19   32.0  Alexander Peartree                   NaN   

                                                title             variety  \
0                   Nicosia 2013 Vulkà Bianco  (Etna)         White Blend   
1       Quinta dos Avidagos 2011 Avidagos Red (Douro)      Portuguese Red   
2       Rainstorm 2013 Pinot Gris (Willamette Valley)          Pinot Gris   
3   St. Julian 2013 Reserve Late Harvest Riesling ...            Riesling   
4   Sweet Cheeks 2012 Vintner's Reserve Wild Child...          Pinot Noir   
5   Tandem 2011 Ars In Vitro Tempranillo-Merlot (N...  Tempranillo-Merlot   
6    Terre di Giurfo 2013 Belsito Frappato (Vittoria)            Frappato   
7               Trimbach 2012 Gewurztraminer (Alsace)      Gewürztraminer   
8   Heinz Eifel 2013 Shine Gewürztraminer (Rheinhe...      Gewürztraminer   
9   Jean-Baptiste Adam 2012 Les Natures Pinot Gris...          Pinot Gris   
10  Kirkland Signature 2011 Mountain Cuvée Caberne...  Cabernet Sauvignon   
11            Leon Beyer 2012 Gewurztraminer (Alsace)      Gewürztraminer   
12  Louis M. Martini 2012 Cabernet Sauvignon (Alex...  Cabernet Sauvignon   
13             Masseria Setteporte 2012 Rosso  (Etna)   Nerello Mascalese   
14           Mirassou 2012 Chardonnay (Central Coast)          Chardonnay   
15        Richard Böc

In [7]:
# Se vuelve a comprobar los valores nulos que se obtienen
missing_values_datadrop = data_drop.isnull().sum() 
missing_values_datadrop

country                     63
description                  0
points                       0
price                     8996
taster_name              26244
taster_twitter_handle    31213
title                        0
variety                      1
winery                       0
dtype: int64

In [8]:
# Se observa que aun tenemos valores nulos en diversas variables, en este siguiente paso se procederá a imputar la feature price
# Se obtiene la media sin decimales, redondeada al valor entero
mean_value = round(data_drop['price'].mean(), 0)
print(mean_value)

35.0


In [9]:
data_drop['price'].fillna(value=mean_value, inplace=True)
print('Dataframe actualizado:\n')
data_drop.head(10)

Dataframe actualizado:



country                                        description  points  price  \
0     Italy  Aromas include tropical fruit, broom, brimston...      87   35.0   
1  Portugal  This is ripe and fruity, a wine that is smooth...      87   15.0   
2        US  Tart and snappy, the flavors of lime flesh and...      87   14.0   
3        US  Pineapple rind, lemon pith and orange blossom ...      87   13.0   
4        US  Much like the regular bottling from 2012, this...      87   65.0   
5     Spain  Blackberry and raspberry aromas show a typical...      87   15.0   
6     Italy  Here's a bright, informal red that opens with ...      87   16.0   
7    France  This dry and restrained wine offers spice in p...      87   24.0   
8   Germany  Savory dried thyme notes accent sunnier flavor...      87   12.0   
9    France  This has great depth of flavor with its fresh ...      87   27.0   

          taster_name taster_twitter_handle  \
0       Kerin O’Keefe          @kerinokeefe   
1          Roger Voss            @vossroger   
2        Paul Gregutt           @paulgwine    
3  Alexander Peartree                   NaN   
4        Paul Gregutt           @paulgwine    
5   Michael Schachner           @wineschach   
6       Kerin O’Keefe          @kerinokeefe   
7          Roger Voss            @vossroger   
8  Anna Lee C. Iijima                   NaN   
9          Roger Voss            @vossroger   

                                               title             variety  \
0                  Nicosia 2013 Vulkà Bianco  (Etna)         White Blend   
1      Quinta dos Avidagos 2011 Avidagos Red (Douro)      Portuguese Red   
2      Rainstorm 2013 Pinot Gris (Willamette Valley)          Pinot Gris   
3  St. Julian 2013 Reserve Late Harvest Riesling ...            Riesling   
4  Sweet Cheeks 2012 Vintner's Reserve Wild Child...          Pinot Noir   
5  Tandem 2011 Ars In Vitro Tempranillo-Merlot (N...  Tempranillo-Merlot   
6   Terre di Giurfo 2013 Belsito Frappato (Vittoria)            Frappato   
7              Trimbach 2012 Gewurztraminer (Alsace)      Gewürztraminer   
8  Heinz Eifel 2013 Shine Gewürztraminer (Rheinhe...      Gewürztraminer   
9  Jean-Baptiste Adam 2012 Les Natures Pinot Gris...          Pinot Gris   

                winery  
0              Nicosia  
1  Quinta dos Avidagos  
2            Rainstorm  
3           St. Julian  
4         Sweet Cheeks  
5               Tandem  
6      Terre di Giurfo  
7             Trimbach  
8          Heinz Eifel  
9   Jean-Baptiste Adam

In [10]:
# Volvemos a comprobar valores nulos
missing_values_datadrop = data_drop.isnull().sum() 
missing_values_datadrop

country                     63
description                  0
points                       0
price                        0
taster_name              26244
taster_twitter_handle    31213
title                        0
variety                      1
winery                       0
dtype: int64

A continuación se procederá a eliminar aquellas filas que tengan valores nulos, ya se advierte que como minimo perderemos 31213 filas, ya que la variable 'taster_twitter_handle', esto se realiza de esta manera porque se pretende utilizar dicha variable para futuras representaciones en tableau si es posible. Tras realizar esta accion tendremos el dataset sin ningún tipo de valor nulo y podremos crear alguna nueva variable para utilizar más adelante en la representación.

In [11]:
# Eliminamos todas aquellas filas que tengan un valor nulo
data_clean = data_drop.dropna()
data_clean.head(10)

country                                        description  points  \
0      Italy  Aromas include tropical fruit, broom, brimston...      87   
1   Portugal  This is ripe and fruity, a wine that is smooth...      87   
2         US  Tart and snappy, the flavors of lime flesh and...      87   
4         US  Much like the regular bottling from 2012, this...      87   
5      Spain  Blackberry and raspberry aromas show a typical...      87   
6      Italy  Here's a bright, informal red that opens with ...      87   
7     France  This dry and restrained wine offers spice in p...      87   
9     France  This has great depth of flavor with its fresh ...      87   
10        US  Soft, supple plum envelopes an oaky structure ...      87   
11    France  This is a dry wine, very spicy, with a tight, ...      87   

    price        taster_name taster_twitter_handle  \
0    35.0      Kerin O’Keefe          @kerinokeefe   
1    15.0         Roger Voss            @vossroger   
2    14.0       Paul Gregutt           @paulgwine    
4    65.0       Paul Gregutt           @paulgwine    
5    15.0  Michael Schachner           @wineschach   
6    16.0      Kerin O’Keefe          @kerinokeefe   
7    24.0         Roger Voss            @vossroger   
9    27.0         Roger Voss            @vossroger   
10   19.0     Virginie Boone               @vboone   
11   30.0         Roger Voss            @vossroger   

                                                title             variety  \
0                   Nicosia 2013 Vulkà Bianco  (Etna)         White Blend   
1       Quinta dos Avidagos 2011 Avidagos Red (Douro)      Portuguese Red   
2       Rainstorm 2013 Pinot Gris (Willamette Valley)          Pinot Gris   
4   Sweet Cheeks 2012 Vintner's Reserve Wild Child...          Pinot Noir   
5   Tandem 2011 Ars In Vitro Tempranillo-Merlot (N...  Tempranillo-Merlot   
6    Terre di Giurfo 2013 Belsito Frappato (Vittoria)            Frappato   
7               Trimbach 2012 Gewurztraminer (Alsace)      Gewürztraminer   
9   Jean-Baptiste Adam 2012 Les Natures Pinot Gris...          Pinot Gris   
10  Kirkland Signature 2011 Mountain Cuvée Caberne...  Cabernet Sauvignon   
11            Leon Beyer 2012 Gewurztraminer (Alsace)      Gewürztraminer   

                 winery  
0               Nicosia  
1   Quinta dos Avidagos  
2             Rainstorm  
4          Sweet Cheeks  
5                Tandem  
6       Terre di Giurfo  
7              Trimbach  
9    Jean-Baptiste Adam  
10   Kirkland Signature  
11           Leon Beyer

In [15]:
# Se comprueba que no existen valores nulos en el dataset
data_clean.isnull().sum()

country                  0
description              0
points                   0
price                    0
taster_name              0
taster_twitter_handle    0
title                    0
variety                  0
winery                   0
dtype: int64

In [45]:
# Se observa la distribución de la variable númerica price

data.price.describe()

count    120975.000000
mean         35.363389
std          41.022218
min           4.000000
25%          17.000000
50%          25.000000
75%          42.000000
max        3300.000000
Name: price, dtype: float64

In [41]:
data_clean.loc[data_clean.price < 10,'price_cat']='Very cheap'
data_clean.loc[(data_clean.price > 10) & (data_clean.price < 25) ,'price_cat']='Cheap'
data_clean.loc[(data_clean.price > 25) & (data_clean.price < 40) ,'price_cat']='Normal'
data_clean.loc[(data_clean.price > 40) & (data_clean.price < 300) ,'price_cat']='Expensive'
data_clean.loc[data_clean.price > 300 ,'price_cat']='Luxury'

In [43]:
data_clean.loc[:,'price_cat']
data_clean.head(10)

country                                        description  points  \
0      Italy  Aromas include tropical fruit, broom, brimston...      87   
1   Portugal  This is ripe and fruity, a wine that is smooth...      87   
2         US  Tart and snappy, the flavors of lime flesh and...      87   
4         US  Much like the regular bottling from 2012, this...      87   
5      Spain  Blackberry and raspberry aromas show a typical...      87   
6      Italy  Here's a bright, informal red that opens with ...      87   
7     France  This dry and restrained wine offers spice in p...      87   
9     France  This has great depth of flavor with its fresh ...      87   
10        US  Soft, supple plum envelopes an oaky structure ...      87   
11    France  This is a dry wine, very spicy, with a tight, ...      87   

    price        taster_name taster_twitter_handle  \
0    35.0      Kerin O’Keefe          @kerinokeefe   
1    15.0         Roger Voss            @vossroger   
2    14.0       Paul Gregutt           @paulgwine    
4    65.0       Paul Gregutt           @paulgwine    
5    15.0  Michael Schachner           @wineschach   
6    16.0      Kerin O’Keefe          @kerinokeefe   
7    24.0         Roger Voss            @vossroger   
9    27.0         Roger Voss            @vossroger   
10   19.0     Virginie Boone               @vboone   
11   30.0         Roger Voss            @vossroger   

                                                title             variety  \
0                   Nicosia 2013 Vulkà Bianco  (Etna)         White Blend   
1       Quinta dos Avidagos 2011 Avidagos Red (Douro)      Portuguese Red   
2       Rainstorm 2013 Pinot Gris (Willamette Valley)          Pinot Gris   
4   Sweet Cheeks 2012 Vintner's Reserve Wild Child...          Pinot Noir   
5   Tandem 2011 Ars In Vitro Tempranillo-Merlot (N...  Tempranillo-Merlot   
6    Terre di Giurfo 2013 Belsito Frappato (Vittoria)            Frappato   
7               Trimbach 2012 Gewurztraminer (Alsace)      Gewürztraminer   
9   Jean-Baptiste Adam 2012 Les Natures Pinot Gris...          Pinot Gris   
10  Kirkland Signature 2011 Mountain Cuvée Caberne...  Cabernet Sauvignon   
11            Leon Beyer 2012 Gewurztraminer (Alsace)      Gewürztraminer   

                 winery  price_cat  
0               Nicosia     Normal  
1   Quinta dos Avidagos      Cheap  
2             Rainstorm      Cheap  
4          Sweet Cheeks  Expensive  
5                Tandem      Cheap  
6       Terre di Giurfo      Cheap  
7              Trimbach      Cheap  
9    Jean-Baptiste Adam     Normal  
10   Kirkland Signature      Cheap  
11           Leon Beyer     Normal

In [47]:
data_clean.points.describe()

count    98695.000000
mean        88.635595
std          2.969586
min         80.000000
25%         87.000000
50%         89.000000
75%         91.000000
max        100.000000
Name: points, dtype: float64

In [50]:
data_clean.loc[data_clean.points < 85,'points_cat'] = 'Not Good'
data_clean.loc[(data_clean.points > 85) & (data_clean.points <= 87) ,'points_cat'] = 'Good'
data_clean.loc[(data_clean.points > 87) & (data_clean.points <= 89) ,'points_cat']='Very good'
data_clean.loc[data_clean.points > 89 ,'points_cat'] = 'Excellent'

In [51]:
data_clean.loc[:,'points_cat']
data_clean.head(10)

country                                        description  points  \
0      Italy  Aromas include tropical fruit, broom, brimston...      87   
1   Portugal  This is ripe and fruity, a wine that is smooth...      87   
2         US  Tart and snappy, the flavors of lime flesh and...      87   
4         US  Much like the regular bottling from 2012, this...      87   
5      Spain  Blackberry and raspberry aromas show a typical...      87   
6      Italy  Here's a bright, informal red that opens with ...      87   
7     France  This dry and restrained wine offers spice in p...      87   
9     France  This has great depth of flavor with its fresh ...      87   
10        US  Soft, supple plum envelopes an oaky structure ...      87   
11    France  This is a dry wine, very spicy, with a tight, ...      87   

    price        taster_name taster_twitter_handle  \
0    35.0      Kerin O’Keefe          @kerinokeefe   
1    15.0         Roger Voss            @vossroger   
2    14.0       Paul Gregutt           @paulgwine    
4    65.0       Paul Gregutt           @paulgwine    
5    15.0  Michael Schachner           @wineschach   
6    16.0      Kerin O’Keefe          @kerinokeefe   
7    24.0         Roger Voss            @vossroger   
9    27.0         Roger Voss            @vossroger   
10   19.0     Virginie Boone               @vboone   
11   30.0         Roger Voss            @vossroger   

                                                title             variety  \
0                   Nicosia 2013 Vulkà Bianco  (Etna)         White Blend   
1       Quinta dos Avidagos 2011 Avidagos Red (Douro)      Portuguese Red   
2       Rainstorm 2013 Pinot Gris (Willamette Valley)          Pinot Gris   
4   Sweet Cheeks 2012 Vintner's Reserve Wild Child...          Pinot Noir   
5   Tandem 2011 Ars In Vitro Tempranillo-Merlot (N...  Tempranillo-Merlot   
6    Terre di Giurfo 2013 Belsito Frappato (Vittoria)            Frappato   
7               Trimbach 2012 Gewurztraminer (Alsace)      Gewürztraminer   
9   Jean-Baptiste Adam 2012 Les Natures Pinot Gris...          Pinot Gris   
10  Kirkland Signature 2011 Mountain Cuvée Caberne...  Cabernet Sauvignon   
11            Leon Beyer 2012 Gewurztraminer (Alsace)      Gewürztraminer   

                 winery  price_cat points_cat  
0               Nicosia     Normal       Good  
1   Quinta dos Avidagos      Cheap       Good  
2             Rainstorm      Cheap       Good  
4          Sweet Cheeks  Expensive       Good  
5                Tandem      Cheap       Good  
6       Terre di Giurfo      Cheap       Good  
7              Trimbach      Cheap       Good  
9    Jean-Baptiste Adam     Normal       Good  
10   Kirkland Signature      Cheap       Good  
11           Leon Beyer     Normal       Good

In [55]:
# Comprobamos que se ha realizado correctamente la categorización

data_clean[1050:1090]

country                                        description  points  \
1378            US  This austere and lightly herbal Cabernet is ba...      86   
1379        France  This is a robust, dense wine, showing the hot ...      86   
1382         Chile  A good but awkward attempt at a bigger, more s...      86   
1384         Italy  This is richer than most Pinot Grigios. It ope...      88   
1386            US  The high alcohol is a good indication of the u...      88   
1387         Italy  This lively white opens with aromas of peach b...      88   
1388            US  Light and clear in color, this wine veers into...      88   
1389  South Africa  The nose of this Chenin is light and dainty, w...      88   
1390  South Africa  An underlying richness and toasty character gr...      88   
1391         Italy  Delicate scents of elderflower and orchard fru...      88   
1392            US  Cola, black cherry and crushed purple flowers ...      88   
1393            US  An uncommon source grape for pink wine, this N...      88   
1394     Argentina  This screaming violet-colored chunkster is big...      88   
1395            US  Rare to find on its own, as it's mainly used i...      88   
1397      Portugal  This LBV from the Bulas family vineyards, down...      88   
1399         Italy  Varietal scents of cut grass, grapefruit and a...      88   
1400      Portugal  This is a fruity, juicy wine that has a dry ta...      88   
1402         Italy  Aromas of orchard fruit, nut and a light whiff...      88   
1403         Italy  This lively white opens with aromas of grapefr...      88   
1404         Italy  Inviting aromas of toasted almond, pear and a ...      88   
1405            US  This is an unusual commingling of mostly Rhone...      88   
1406            US  This big but balanced wine has melon and apple...      88   
1407         Italy  Grapefruit, peach and nectarine aromas carry o...      88   
1408         Italy  This linear wine opens with aromas that sugges...      88   
1409            US  Opaque and thick, this soft, black peppery win...      88   
1411      Portugal  The dark color and high acidity of this wine a...      88   
1412      Portugal  The balance between wood aging and fruit is at...      88   
1413            US  While big and brawny in style, this Pinot stil...      88   
1414        France  Big, full bodied and concentrated, this is a r...      93   
1415        France  This is a remarkably spicy wine that is also f...      93   
1416        France  This is a medium-sweet wine that's warm and ri...      93   
1418            US  The sixteenth edition of Dunham's benchmark Ca...      93   
1419            US  Richly aromatic, on the strength of the nose a...      93   
1420            US  Sourced entirely from the Sagemoor vineyard, t...      93   
1421        France  Sweet and rich, this is a wine that combines h...      93   
1422       Austria  Alluring white-pepper spice, with nuances of d...      93   
1424            US  Sourced principally from Klipsun, with smaller...      92   
1425       Austria  Red cherry fruit, with edges of white pepper, ...      92   
1426        France  This is a textured wine with ripe and perfumed...      92   
1427        France  Richly sweet, this is a wine that has traded p...      92   

      price        taster_name taster_twitter_handle  \
1378   18.0       Paul Gregutt           @paulgwine    
1379   24.0     Joe Czerwinski                @JoeCz   
1382   20.0  Michael Schachner           @wineschach   
1384   25.0      Kerin O’Keefe          @kerinokeefe   
1386   36.0       Paul Gregutt           @paulgwine    
1387   27.0      Kerin O’Keefe          @kerinokeefe   
1388   42.0     Virginie Boone               @vboone   
1389   16.0      Lauren Buzzeo             @laurbuzz   
1390   35.0      Lauren Buzzeo             @laurbuzz   
1391   20.0      Kerin O’Keefe          @kerinokeefe   
1392   48.0      Matt Kettmann         @mattkettmann   
1393   35.0      

Tras realizar toda la limpieza de datos, la imputación de algunos valores nulos más la elimincación de aquellas filas vacias, se obtiene un dataset totalmente 'lleno', con el que es posible realizar representaciones elaboradas, por tanto se procederá a exportar el csv para comenzar dicho trabajo.

In [57]:
data_clean.to_csv('data_clean.csv', index=False)